replicatedhq/ship
# Improve error messaging around Github fetching

Fixes #421 

What I Did
------------
Add a check that a base path was found on extraction of a Github archive

How I Did it
------------
Check if a file or folder prefixed with `basePath` was found. If none are found, throw an error that the path is not found in the repository.

How to verify it
------------
Run `ship init github.com/helm/charts/mysql` and you should see this error:
```
There was an unexpected error! resolve release: determine type of github.com/helm/charts/mysql: fetch repo contents: Path mysql in helm/charts on branch master not found
```

Description for the Changelog
------------
- Improve error messaging around Github fetching


Picture of a Boat (not required but encouraged)
------------
![](http://4.bp.blogspot.com/-zpDJECfuD_A/VR2pTl0kbGI/AAAAAAAAC9A/Vdp5_3fRYSc/s1600/korra3x10_1360.jpg)
